pallets/ajuna-tournament/src/lib.rs

```diff
@@ -682,37 +823,25 @@ pub mod pallet {
 		}
 	}
 
-	impl<T: Config<I>, I: 'static> TournamentRanker<T::SeasonId, T::RankedEntity, T::EntityId>
-		for Pallet<T, I>
+	impl<T: Config<I>, I: 'static>
+		TournamentRanker<TournamentCategoryIdFor<T, I>, T::RankedEntity, T::EntityId> for Pallet<T, I>
 	{
-		fn try_rank_entity_in_tournament_for<R>(
-			season_id: &T::SeasonId,
+		fn try_rank_entity_in_tournament_for(
+			category_id: &TournamentCategoryIdFor<T, I>,
 			entity_id: &T::EntityId,
 			entity: &T::RankedEntity,
-			ranker: &R,
-		) -> DispatchResult
-		where
-			R: EntityRank<EntityId = T::EntityId, Entity = T::RankedEntity>,
-		{
-			if !ranker.can_rank((entity_id, entity)) {
+		) -> DispatchResult {
+			let (tournament_id, tournament_config) =
+				Self::get_active_tournament_config_for(category_id)
+					.ok_or::<Error<T, I>>(Error::<T, I>::NoActiveTournamentForCategory)?;
+
+			if !tournament_config.ranker.can_rank((entity_id, entity)) {
 				return Ok(());
 			}
 
-			ensure!(
-				matches!(
-					Self::get_active_tournament_state_for(season_id),
-					TournamentState::ActivePeriod(_)
-				),
-				Error::<T, I>::NoActiveTournamentForSeason
-			);
-
-			let (tournament_id, tournament_config) =
-				Self::get_active_tournament_config_for(season_id)
-					.ok_or::<Error<T, I>>(Error::<T, I>::TournamentNotFound)?;
-
```

Comment:
Before I used the `get_active_tournament_state_for` to avoid calling `get_active_tournament_config` if the Tournament was not in an Active state, but now that the ranker is part of the tournament_config its more efficient to just call `get_active_tournament_config` directly

Reply:
Cool thanks for the reply!

Reply:
Taking that into account I noticed that in both `try_claim*` we do an unnecessary check with `get_active_tournament_state`. So I removed that check